android/build.gradle

```diff
@@ -368,16 +366,56 @@ task createNativeDepsDirectories(dependsOn: applyJavaPatches) {
     thirdPartyNdkDir.mkdirs()
 }
 
-task downloadBoost(dependsOn: createNativeDepsDirectories, type: Download) {
+def resolveTaskFactory(String name, String artifactLocalName, File reactNativeAndroidDownloadDir, File reanimatedDownloadDir) {
+    return tasks.create(name: name.startsWith("resolve") ? name : "resolve" + name, dependsOn: createNativeDepsDirectories, type: Copy) {
+        from reactNativeAndroidDownloadDir
+        include artifactLocalName
+        into reanimatedDownloadDir
+
+        onlyIf {
+            // First we check whether the file is already in our download directory
+            if (file("$reanimatedDownloadDir/$artifactLocalName").isFile()) {
+                return false
+            }
+
+            // If it is not the case we check whether it was downloaded by ReactAndroid project
+            def tarFile = Paths.get(reactNativeAndroidDownloadDir.toString(), artifactLocalName).toFile()
+            return tarFile.isFile()
+        }
+    }
+}
+
+Task resolveBoost = resolveTaskFactory("resolveBoost", "boost_${BOOST_VERSION}.tar.gz", reactNativeAndroidDownloadDir, downloadsDir)
+Task resolveDoubleConversion = resolveTaskFactory("resolveDoubleConversion", "double-conversion-${DOUBLE_CONVERSION_VERSION}.tar.gz", reactNativeAndroidDownloadDir, downloadsDir)
+Task resolveFolly = resolveTaskFactory("resolveFolly", "folly-${FOLLY_VERSION}.tar.gz", reactNativeAndroidDownloadDir, downloadsDir)
+Task resolveGlog = resolveTaskFactory("resolveGlog", "glog-${GLOG_VERSION}.tar.gz", reactNativeAndroidDownloadDir, downloadsDir)
+
+def reactNativeAndroidProject = findProject(":ReactAndroid")
+if (reactNativeAndroidProject != null) {
+    reactNativeAndroidProject.afterEvaluate {
+        def resolveTasks = [resolveBoost, resolveGlog, resolveDoubleConversion, resolveFolly]
+        resolveTasks.forEach({ t ->
+            def reactAndroidDownloadTask = reactNativeAndroidProject.getTasks().findByName("download" + t.name.replace("resolve", ""))
+            if (reactAndroidDownloadTask != null) {
```

Comment:
I think we could remove this `null` check here (because these tasks should exist) but who knows what 0.71 brings us (prefabs)... 🤷

Reply:
I think it might be fine to throw exception if we wanted to get notified that some behaviour in React Native's internals changed. If it is the case let's make it an explicit Exception with reasonable message instead of NPE. 

Otherwise I would rather leave it this way.

Reply:
How about this: [d737b35](https://github.com/software-mansion/react-native-reanimated/pull/3588/commits/d737b35152df865270fa7d05dc0ba953e33879f9)? 

```groovy
String reactAndroidDownloadTaskName = "download" + task.name.replace("resolve", "")
def reactAndroidDownloadTask = reactNativeAndroidProject.getTasks().findByName(reactAndroidDownloadTaskName)
if (reactAndroidDownloadTask != null) {
    task.dependsOn(reactAndroidDownloadTask)
} else {
     throw new GradleScriptException("[Reanimated] Failed to find task named `$reactAndroidDownloadTaskName` in `:ReactAndroid` project." +
        " Explicit dependency between it and $task.name task can not be set.")
}
```

We could also consider just using `logger.warn` instead of throwing as it does not have to stop the build from succeeding.

Reply:
I think `logger.warn` is a better choice indeed.

Reply:
[345a009](https://github.com/software-mansion/react-native-reanimated/pull/3588/commits/345a0091598dbd76dff393cd6c9d857065da331d)